src/Stripe.Tests.XUnit/coupons_new/coupons.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class coupon_test
+    {
+        private string previousApiBase;
+        private readonly string resourceId = "co_123";
+        private StripeCouponService service;
+
+        public coupon_test()
+        {
+            this.service = new StripeCouponService(Cache.ApiKey);
+            this.previousApiBase = StripeConfiguration.GetApiBase();
+            StripeConfiguration.SetApiBase("http://localhost:12111/v1");
+        }
+
+        ~coupon_test()
+        {
+            StripeConfiguration.SetApiBase(this.previousApiBase);
+        }
+
+        [Fact]
+        public void create_succeeds()
+        {
+            var options = new StripeCouponCreateOptions() {
+                // Add a space at the end to ensure the ID is properly URL encoded
+                // when passed in the URL for other methods
```

Comment:
OMG, this comment: I see that this looks copied from elsewhere, but does anyone know what exactly causes this problem?

Reply:
@brandur-stripe it's mostly because some endpoint allow you to pass user-chosen ids and so we had encoding issues. We should re-build the encoding logic to do this later in the pipeline though we haven't scheduled the work.
I'm not planning to keep it, I was really mostly reproducing all the tests for the normal coupon package.

Reply:
I think Brandur was confused like I was: the comment implies that we _have_ to add a space to work around some URL-encoding problem, not that we're adding a space specifically to test that URL-encoding works as expected. In any case, URL-encoding correctness should be tested separately.

Reply:
yep but at the time that was the easiest way to make sure we did not break it again. Totally happy to remove this once we agree this shape of test is the right direction!

Reply:
Oh man, yep, I was confused by that comment.